ovos_utils/gui.py

```diff
@@ -752,7 +753,19 @@ def show_pages(self, page_names, index=0, override_idle=None,
         data = self.__session_data.copy()
         data.update({'__from': self.skill_id})
         self.bus.emit(Message("gui.value.set", data))
+
+        # upload qml contents directly to support gui running in different devices
         page_urls = self._pages2uri(page_names)
+        contents = []
+        for p in page_urls:
```

Comment:
Should this be conditional or happen during init instead of runtime? This will add a bunch of extra runtime parsing that will definitely be measurable on a Mark2

Reply:
A config option to upload QML (default to False) would handle this in a backwards-compatible way